src/coreclr/tools/aot/ILCompiler/ILCompilerRootCommand.cs

```diff
@@ -135,7 +135,7 @@ internal sealed class ILCompilerRootCommand : RootCommand
         public Option<string[]> DirectPInvokeLists { get; } =
             new(new[] { "--directpinvokelist" }, Array.Empty<string>, "File with list of PInvokes to call directly");
         public Option<int> MaxGenericCycle { get; } =
-            new(new[] { "--maxgenericcycle" }, () => CompilerTypeSystemContext.DefaultGenericCycleCutoffPoint, "Max depth of generic cycle");
+            new(new[] { "--maxgenericcycle" }, () => CompilerTypeSystemContext.DefaultGenericCycleDepthCutoff, "Max depth of generic cycle");
```

Comment:
Could you add a new argument to control the other cutoff? If I'm reading it right, the new cutoff also applies to NativeAOT and we should have a way to control it if it misbehaves.

I don't have a good mental model of how the new cutoff will work in practice - maybe we should also have a `-1` value that means "don't check for this".

Reply:
Added in 8th commit.